src/ServerRequest.php

```diff
@@ -180,23 +180,35 @@ public static function fromGlobals()
             ->withUploadedFiles(self::normalizeFiles($_FILES));
     }
 
+    private static function extractPortFromHost($host)
+    {
+        $url = "http://{$host}";
+        $parts = parse_url($url);
+        $host = $parts['host'];
```

Comment:
`parse_url` returns false if the resulting uri is invalid, e.g. the host is empty. this would then fail. so `$_SERVER['HTTP_HOST']` should be skipped in this case IMO